### PR TITLE
Intercom overhaul

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do
Touches up on Intercom code and sprites, as well as giving their sprites more details about what is going on with them. It uses the old wallradio sprites into the new intercom sprites, with slight changes. Last deconstruction step now requires a wrench instead of a welder.

## Why It's Good For The Game
It gives a visual description of what is going on with an intercom, often you don't know that a certain intercom is set to a specific setting (like the AI Private intercoms in captains office and in AI upload, or the command one in the Meeting Room). This also gives a specific purpose to the Malf AI lipreading skill because you can get away pretty well with only intercomms currently. Also consistency is good for deconstruction.

## Images of changes

image
(left: different intercomm settings for muted, receiving, transmitting, matching to the settings in the UI.)
(right: the icons for all the channels: common, AI private, response team, command, engi, medical, security, science, supply, service, proceedure, and syndicate comms.)

## Testing

Tested deconstruction, reconstruction, checked changing various settings, all working as intended
image
Checked various settings around the station, all appearing as they should
(Captains office and AI upload)
image
(AI sat)
image
(Magi's office)
image

## Changelog
:cl:
tweak: Intercoms now show their channel and status on their sprites
imageadd: Intercom channels and recieving/transmitting statuses
imageadd: New intercom sprites
tweak: Intercom's last deconstruction step has be turned into a wrench act to be consistent with fire alarms.
/:cl:

### CODE REVIEW

**Comment by Contrabang** [comment]
Some more examples showing where this is useful
image
image
image

**Review by Sirryan2002 — COMMENTED** [review]
just stuff that popped out immediately to me

File: code/game/objects/items/devices/radio/intercom.dm [hunk 1]
```diff
@@ -141,98 +155,154 @@
...
-	buildstage = 0
+	new /obj/item/intercom_electronics(loc)
+	to_chat(user, "<span class='notice'>You pry out the circuit board from [src].</span>")
+	buildstage = INTERCOM_FRAME
+	update_icon(UPDATE_ICON)
 
 /obj/item/radio/intercom/screwdriver_act(mob/user, obj/item/I)
-	if(buildstage != 2)
-		return ..()
+	if(!(buildstage == INTERCOM_READY))
```
[the remark this answers is not in the document]
Comment (Contrabang) [reply code/game/objects/items/devices/radio/intercom.dm:218]: Done

File: code/game/objects/items/devices/radio/intercom.dm [hunk 2]
```diff
@@ -141,98 +155,154 @@
...
-	update_icon(UPDATE_ICON_STATE | UPDATE_OVERLAYS)
-	update_operating_status(FALSE)
+	wiresexposed = TRUE
+	buildstage = INTERCOM_UNWIRED
+	update_icon(UPDATE_ICON)
 
-/obj/item/radio/intercom/welder_act(mob/user, obj/item/I)
-	if(buildstage != 0)
+/obj/item/radio/intercom/wrench_act(mob/user, obj/item/I)
+	if(!(buildstage == INTERCOM_FRAME))
```
[the remark this answers is not in the document]
Comment (Contrabang) [reply code/game/objects/items/devices/radio/intercom.dm:250]: Done

**Comment by meow20** [comment]
I love everything about it except for it messing with AI's that eaavesdrop on the crew whether malf or not.

**Comment by ItsMarmite** [comment]
Really not a fan of it showing AI private. Feels like an indirect nerf to malf and subverted AIs.

**Comment by Contrabang** [comment]
Most people don't seem to like this, may have a better idea on the future but this won't work

**Pull request closed without merging**